@here/harp-utils/lib/ObjectUtils.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Deep clone of object.
+ *
+ * Like `JSON.parse(JSON.stringify(obj))`, but supports basic javascript types (string, number,
+ * object), `Date` and `RegExp`s and cycles.
+ *
+ * Throws error if enounters object with `prototype` assuming that in general class instances
+ * cannot be reliably cloned by generic algorithm.
+ */
+export function cloneDeep<T>(obj: T): T {
+    const cache: Map<object, object> = new Map();
+    function cloneInternal(src: any): any {
+        if (src === null) {
+            return null;
+        } else if (typeof src === "object") {
+            const cached = cache.get(src);
+            if (cached !== undefined) {
+                return cached;
+            }
+
+            if (Array.isArray(src)) {
+                const result: any[] = [];
+                cache.set(src, result);
+                result.length = src.length;
+                for (let i = 0; i < result.length; ++i) {
+                    result[i] = cloneInternal(src[i]);
+                }
+                return result;
+            } else if (src.constructor === Date) {
+                const result = new Date(src.getTime());
+                cache.set(src, result);
+                return result;
+            } else if (src.constructor === RegExp) {
+                const result = new Date(src.source, src.flags);
```

Comment:
new RegEx.. maybe it's better if you check with `instanceof` instead of comparing the prototype